Assignment_1/main.swift

```diff
@@ -0,0 +1,267 @@
+
+//
+//  main.swift
+//  assignment1
+//
+//  Created by Sanskar on 11/01/22.
+//
+import Foundation
+
+protocol Tax
+{
+    static func calculateTax(price : Double) -> Double
+}
+
+class RawTax : Tax
+{
+    static func calculateTax(price : Double) -> Double
+    {
+        return (0.125 * price)
+    }
+}
+
+class ManufacturedTax : Tax
+{
+    static func calculateTax(price : Double) -> Double
+    {
+        return (0.125 * price + (0.02 * (price + (0.125 * price))))
+    }
+}
+
+class ImportedTax : Tax
+{
+    static func calculateTax(price : Double) -> Double
+    {
+        let finalCost : Double = (0.01 * price) + price
+        if( finalCost > 200)
+        {
+            return (0.05 * finalCost)
+        }
+        else if( finalCost > 100)
+        {
+            return (finalCost + 10)
+        }
+        return (finalCost + 5)
+    }
+}
+
+class userDetail
+{
+    private let name: String?
+    private let quantity: Int
+    private let price: Double?
+    private let type: String?
+    
+    init(_ name:String,_ quantity:Int,_ price:Double,_ type:String)
+    {
+        self.name = name
+        self.quantity = quantity
+        self.price = price
+        self.type = type
+    }
+    
+    func getTotalPrice(tax : Double) -> Double
+    {
+        return (tax + self.price!)
+    }
+    
+    func getOverAllPrice(tax : Double) -> Double
```

Comment:
you should not be needing these functions in userDetail class. it should only have number of objects possessed by user

Reply:
done